### PR TITLE
feat(route): 掘金 开发者活动

### PR DESCRIPTION
**Pull request by moke8**
## 该 PR 相关 Issue / Involved issue

Close #

## 完整路由地址 / Example for the proposed route(s)

```routes
/juejin/events/:city?
```

## 新 RSS 检查列表 / New RSS Script Checklist
  
- [x] 新的路由 New Route
  - [x]  跟随 [v2 路由规范](https://docs.rsshub.app/joinus/script-standard.html) Follows [v2 Script Standard](https://docs.rsshub.app/en/joinus/script-standard.html)
- [x]  文档说明 Documentation
  - [x]  中文文档 CN
  - [ ] 英文文档 EN
- [ ] 全文获取 fulltext
  - [ ] 使用缓存 Use Cache
- [ ] 反爬/频率限制 anti-bot or rate limit?
  - [ ] 如果有, 是否有对应的措施? If yes, do your code reflect this sign?
- [ ] [日期和时间](https://docs.rsshub.app/joinus/pub-date.html) [date and time](https://docs.rsshub.app/en/joinus/pub-date.html)
  - [ ] 可以解析 Parsed
  - [ ] 时区调整 Correct TimeZone
- [ ] 添加了新的包 New package added
- [ ] `Puppeteer`

### CODE REVIEW

File: docs/programming.md [hunk 1]
```diff
@@ -925,6 +925,10 @@ GitHub 官方也提供了一些 RSS:
 
 <Route author="cancergary" example="/juejin/news/739332228916791" path="/juejin/news/:userId" :paramsDesc="['用户 id, 可在用户页 URL 中找到']" radar="1" rssbud="1"/>
 
+### 开发者活动
+
+<Route author="moke8" example="/juejin/events/shanghai" path="/juejin/events/:city" :paramsDesc="['活动地点, 可选, 一般为城市名全拼如shanghai, 支持城市可在掘金活动页查阅']" radar="1" rssbud="1"/>
```
Comment (TonyRL) [inline docs/programming.md:930]:
```suggestion
<Route author="moke8" example="/juejin/events/shanghai" path="/juejin/events/:city?" :paramsDesc="['活动地点, 可选, 一般为城市名全拼如shanghai, 支持城市可在掘金活动页查阅']" radar="1" rssbud="1"/>
```
Path here must match `maintainer.js` and `router.js`

File: lib/v2/juejin/events.js [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+            city: city === 'all' ? '' : city,
+            count: 20,
+            cursor: 0,
+        },
+    });
+
+    const items = response.data.data.map((item) => {
+        const content = item.title;
+        const title = content;
+        const guid = item.id;
```
Comment (TonyRL) [inline lib/v2/juejin/events.js:20]: This guid is way too easy for collision
You can use something like
```suggestion
        const guid = `juejin:events:${city ? `${city}:` : ''}${item.id}`;
```
instead

File: lib/v2/juejin/events.js [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+        const link = item.url;
+        const imgs = `<img src="${item.screenshot}">`;
+        const description = `
+            ${content.replace(/\n/g, '<br>')}<br>
+            ${imgs}<br>
+        `;
```
Comment (TonyRL) [inline lib/v2/juejin/events.js:26]: Please use [art-template](https://docs.rsshub.app/joinus/script-standard.html#template)

File: lib/v2/juejin/events.js [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+            title,
+            link,
+            description,
+            guid,
+        };
+    });
+
+    ctx.state.data = {
+        title: `开发者活动 ${cityName}`,
+        link: `https://juejin.cn/events/${encodeURIComponent(city)}`,
```
Comment (TonyRL) [inline lib/v2/juejin/events.js:40]:
```suggestion
        link: `https://juejin.cn/events/${city}`,
```
No need to encode plain alphabets

File: lib/v2/juejin/maintainer.js [hunk 5]
```diff
@@ -10,4 +10,5 @@ module.exports = {
     '/shares/:userId': ['qiwihui'],
     '/tag/:tag': ['isheng5'],
     '/trending/:category/:type': ['moaix'],
+    '/events/:city?': ['moke8'],
```
Comment (TonyRL) [inline lib/v2/juejin/maintainer.js:13]: Please insert new entries in alphabetical order

File: lib/v2/juejin/router.js [hunk 6]
```diff
@@ -10,4 +10,5 @@ module.exports = (router) => {
     router.get('/shares/:userId', require('./shares'));
     router.get('/tag/:tag', require('./tag'));
     router.get('/trending/:category/:type', require('./trending'));
+    router.get('/events/:city?', require('./events'));
```
Comment (TonyRL) [inline lib/v2/juejin/router.js:13]: Please insert new entries in alphabetical order

File: lib/v2/juejin/events.js [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+module.exports = async (ctx) => {
+    const city = ctx.params.city === 'all' ? '' : ctx.params.city;
+    let cityName = '';
+
+    const response = await got({
+        method: 'post',
+        url: 'https://api.juejin.cn/event_api/v1/event/event_list',
+        json: {
+            city: city === 'all' ? '' : city,
+            count: 20,
```
Comment (TonyRL) [inline lib/v2/juejin/events.js:12]:
```suggestion
            count: ctx.query.limit ? parseInt(ctx.query.limit) : 20,
```
This way users can specify the number of items they want using common parameters

**Comment by moke8** [comment]
机器人报的问题不是我提交的代码

**Review by TonyRL — COMMENTED** [review]
Please take a look at **my** review instead, and remove any changes on `yarn.lock`

**Comment by DIYgod** [comment]
This PR has been open for too long and will be closed first. If you have any questions, feel free to continue the discussion below.

**Pull request closed without merging**